src/app/onboarding/roles/page.tsx

```diff
@@ -1,57 +1,130 @@
 
-const roleOptions = new Map<RoleEnum, string>([
+const roleOptions = new Map<string, string>([
   ['ATTORNEY', 'Attorney'],
   ['INTERPRETER', 'Interpreter'],
+  ['LEGAL_FELLOW', 'Legal Fellow'],
+  ['ATTORNEY,INTERPRETER', 'Attorney and Interpreter'],
+  ['LEGAL_FELLOW,INTERPRETER', 'Legal Fellow and Interpreter'],
 ]);
 
+const roleSchema = z.object({
+  roles: z
+    .enum([
+      'ATTORNEY',
+      'INTERPRETER',
+      'LEGAL_FELLOW',
+      'ATTORNEY,INTERPRETER',
+      'LEGAL_FELLOW,INTERPRETER',
+      '',
+    ])
+    .superRefine((input, ctx) => {
+      if (input === '')
+        ctx.addIssue({
+          code: 'custom',
+          message: 'Must include at least one role',
+        });
+      return ctx;
+    }),
+});
+
 export default function Page() {
   const onboarding = useContext(OnboardingContext);
+  const { push } = useRouter();
 
-  const setRoles = (roles: Set<RoleEnum>) => {
-    onboarding?.setRoles(roles);
-    if (roles.size !== 0) {
-      onboarding?.setCanContinue(true);
-    } else {
-      onboarding?.setProgress(0);
-      onboarding?.setCanContinue(false);
-    }
+  const onSubmit = (values: z.infer<typeof roleSchema>) => {
+    if (!onboarding) throw new Error('Fatal: no onboarding layout detected');
 
-    if (roles.has('ATTORNEY')) {
-      onboarding?.setFlow([
-        { name: 'Roles', url: 'roles' },
-        { name: 'Basic Info', url: 'basic-information' },
-        { name: 'Availability', url: 'availability' },
-        { name: 'Legal Experience', url: 'legal-experience' },
-        { name: 'Done', url: 'done' },
-      ]);
+    const roles = values.roles.split(',') as RoleEnum[];
+    onboarding.setRoles(roles);
+
+    let newFlow: FlowData[];
+
+    if (roles.includes('ATTORNEY')) {
+      newFlow = ATTORNEY_FLOW;
+    } else if (roles.includes('LEGAL_FELLOW')) {
+      newFlow = LEGAL_FELLOW_FLOW;
     } else {
-      onboarding?.setFlow([
-        { name: 'Roles', url: 'roles' },
-        { name: 'Basic Info', url: 'basic-information' },
-        { name: 'Availability', url: 'availability' },
-        { name: 'Done', url: 'done' },
-      ]);
+      newFlow = INTERPRETER_FLOW;
     }
+
+    onboarding.setFlow(newFlow);
+    push(`/onboarding/${newFlow[1].url}`);
   };
 
+  const form = useForm<z.infer<typeof roleSchema>>({
+    resolver: zodResolver(roleSchema),
+    defaultValues: {
+      roles: (onboarding
+        ? Array.from(onboarding.roles.values() || []).join(',')
+        : '') as RoleOptionType,
+    },
+  });
+
   return (
-    <>
-      <H1>Roles</H1>
-
-      <InputDropdown
-        label="What role(s) would you like to hold?"
-        multi
-        defaultValue={onboarding?.roles}
-        onChange={v => setRoles(v as Set<RoleEnum>)}
-        options={roleOptions}
-      />
-    </>
+    <FormProvider {...form}>
+      {/* div to fill top space, delete after nav bar is added */}
+      <div />
+      <SmallCardForm onSubmit={form.handleSubmit(onSubmit)}>
+        <H1>Role</H1>
```

Comment:
have we decided whether we want the nav bar on the Roles page, if it isn't there for all other onboarding pages? Figma still includes the nav bar on the Roles page. @kyrenetam

Reply:
Let's discuss on Sunday

Reply:
from worksesh: yes to navbar in roles